### PR TITLE
Course maintenance: Ask for course code before deleting *or* recycling #3302

### PR DESCRIPTION
**Pull request by carlangas159**

### CODE REVIEW

**Comment by AngelFQC** [comment]
Moviendo PR de master a 1.11.x

File: main/course_info/delete_course.php [hunk 1]
```diff
@@ -40,14 +40,27 @@
     $message .= '<p>'.get_lang('ByDel').'</p>';
     $message .= '<p><a class="btn btn-primary" 
         href="'.api_get_path(WEB_CODE_PATH).'course_info/maintenance.php?'.api_get_cidreq().'">'.
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
+        get_lang('No').'</a>&nbsp;<a class="btn btn-danger delete-course">'.
```
Comment (AngelFQC) [inline main/course_info/delete_course.php:43]: Para este caso, tener un `a` sin `href`es mejor cambiarlo por un `button`

File: main/course_info/delete_course.php [hunk 2]
```diff
@@ -38,16 +38,29 @@
 } else {
     $message = '<h3>'.get_lang('Course').' : '.$current_course_name.' ('.$current_course_code.') </h3>';
     $message .= '<p>'.get_lang('ByDel').'</p>';
-    $message .= '<p><a class="btn btn-primary" 
+    $message .= '<p><a class="btn btn-primary"
         href="'.api_get_path(WEB_CODE_PATH).'course_info/maintenance.php?'.api_get_cidreq().'">'.
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
-        get_lang('Yes').'</a></p>';
+        get_lang('No').'</a>&nbsp;<buttom class="btn btn-danger delete-course">'.
```
Comment (ywarnier) [inline main/course_info/delete_course.php:43]: button, not buttom

**Comment by ywarnier** [comment]
In case you don't really *need* JavaScript, for user experience reasons, please use PHP directly.

In this case, you can add the course code form element to the page itself (withouth JS) before the buttons, and change the title of the buttons a bit, like you would find on similar forms in AWS or DigitalOcean.

The "no" button could be changed to a link to simply go back to the previous page.

Essentially, your PR does the job but in an excessively JS-dependent manner. Although having JS support is a requirement now in Chamilo, it does reduce accessibility for things that could very well be handled fully in HTML (in this case)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5709378&bid=18684644)

File: main/course_info/delete_course.php [hunk 3]
```diff
@@ -25,29 +25,49 @@
...
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
-        get_lang('Yes').'</a></p>';
+    $message = '<h3>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h3>';
+    $message .= '<p>' . get_lang('ByDel') . '</p>';
+    $message .= '<p><span class="form_required">*</span>' . get_lang('CourseCodeConfirmation') . '&nbsp;<input type="text" name="course_code" id="course_code"></p>';
+
+    $message .= '<p>';
+    $message .= '<button class="btn btn-danger delete-course">' . get_lang('ValidateChanges') . '</button>';
+    $message .= '&nbsp;';
+    $message .= '<a class="btn btn-primary"href="' . api_get_path(WEB_CODE_PATH) . 'course_info/maintenance.php?' . api_get_cidreq() . '">' .
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:54]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 141 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5709378)

File: main/course_info/delete_course.php [hunk 4]
```diff
@@ -25,29 +25,49 @@
...
+        Session::erase('_cid');
+        Session::erase('_real_cid');
+        $message = '<h2>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h2>';
+        $message .= get_lang('HasDel');
+        $message .= '<br /><br /><a href="../../index.php">' . get_lang('BackHome') . '</a>';
+    } else {
+        /* message if code course is incorrect */
+        $message = '<h2>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h2>';
+        $message .= '<p>'.get_lang('CourseRegistrationCodeIncorrect').'</p>';
+        $message .= '<p><a class="btn btn-primary" href="' . api_get_path(WEB_CODE_PATH) . 'course_info/delete_course.php?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a>';
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:42]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 192 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5709378)

File: main/course_info/delete_course.php [hunk 5]
```diff
@@ -25,29 +25,49 @@
...
 } else {
-    $message = '<h3>'.get_lang('Course').' : '.$current_course_name.' ('.$current_course_code.') </h3>';
-    $message .= '<p>'.get_lang('ByDel').'</p>';
-    $message .= '<p><a class="btn btn-primary" 
-        href="'.api_get_path(WEB_CODE_PATH).'course_info/maintenance.php?'.api_get_cidreq().'">'.
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
-        get_lang('Yes').'</a></p>';
+    $message = '<h3>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h3>';
+    $message .= '<p>' . get_lang('ByDel') . '</p>';
+    $message .= '<p><span class="form_required">*</span>' . get_lang('CourseCodeConfirmation') . '&nbsp;<input type="text" name="course_code" id="course_code"></p>';
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:49]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 165 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5709378)

**Pull request closed without merging**